src/main/java/duke/command/Parser.java

```diff
@@ -0,0 +1,112 @@
+package duke.command;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+
+/**
+ * Parser to parse strings into commands.
+ */
+public class Parser {
+
+    /**
+     * Parses a string into a command.
+     *
+     * @param input Input string to parse into command.
+     * @return Parsed command.
+     */
+    public static Command parse(String input)
+            throws IllegalArgumentException {
+        LinkedHashMap<Command.Argument, String> arguments = new LinkedHashMap<>();
+        for (String term : input.strip().split(" /")) {
+            int firstSpace = term.indexOf(" ");
+            Command.Argument argument = Command.parseArgument((firstSpace == -1
```

Comment:
I am not too sure about this but
···
alpha = (aLongBooleanExpression) ? beta : gamma;
alpha = (aLongBooleanExpression)
? beta
: gamma;
···
so should the Boolean be bracketed?